custom_components/elmo_iess_alarm/binary_sensor.py

```diff
@@ -32,17 +34,38 @@ async def async_setup_entry(
     inventory = await hass.async_add_executor_job(device._connection._get_descriptions)
     for sector_id, name in inventory[query.SECTORS].items():
         unique_id = f"{entry.entry_id}_{DOMAIN}_{query.SECTORS}_{sector_id}"
-        sensors.append(EconnectDoorWindowSensor(coordinator, device, unique_id, sector_id, query.SECTORS, name))
+        
+    # Check if there is an alias in configuration flow and use it for naming
```

Comment:
I think this code works, but it's actually repeated multiple times. What if you write a dedicated function in the `helpers.py` module (https://github.com/palazzem/ha-econnect-alarm/blob/e30240ab4c2125496b1f14dd6f8839eb52a23ba4/custom_components/elmo_iess_alarm/helpers.py)?

So you add a function like `generate_entity_name ` or `get_entity_name` (or anything you would like to propose!) that has includes your code. We put all the logic to define the entity name there and we test it. Then you simply replace all these code-blocks with the function name.

If you write the function, don't worry about writing the `docstring`. In general I use ChatGPT to generate docstrings using the Google Python coding guidelines. I can write one for you once the PR is almost ready!